test/e2e-secondary-network/secondary_network_test.go

```diff
@@ -219,6 +223,221 @@ func (data *testData) pingBetweenInterfaces(t *testing.T) error {
+			}
+
+			released := true
+			for _, ipAddress := range ipPool.Status.IPAddresses {
+				if podIPString == ipAddress.IPAddress {
+					released = false
+					break
+				}
+			}
+
+			if released {
+				return nil
+			}
+		}
+	}
+}
+
+// getNetAttachDefClient returns a NetAttachDef client.
+func (data *testData) getNetAttachDefClient() (netattdef.Interface, error) {
+	config, err := data.e2eTestData.GetKubeConfig()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get kubeconfig: %v", err)
+	}
+
+	client, err := netattdef.NewForConfig(config)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create NetAttachDef client: %v", err)
+	}
+	return client, nil
+}
+
+// getIPPoolNames retrieves the IPPool names from a NetworkAttachmentDefinition.
+func (data *testData) getIPPoolNames(networkName, namespace string) ([]string, error) {
+	client, err := data.getNetAttachDefClient()
+	if err != nil {
+		return nil, err
+	}
+
+	nad, err := client.K8sCniCncfIoV1().NetworkAttachmentDefinitions(namespace).Get(context.TODO(), networkName, metav1.GetOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("failed to get NetworkAttachmentDefinition %s: %v", networkName, err)
+	}
+
+	var nadConfig struct {
+		IPAM struct {
+			IPPools []string `json:"ippools"`
+		} `json:"ipam"`
+	}
+
+	if err := json.Unmarshal([]byte(nad.Spec.Config), &nadConfig); err != nil {
+		return nil, fmt.Errorf("failed to parse NAD config JSON: %v", err)
+	}
+
+	return nadConfig.IPAM.IPPools, nil
+}
+
+// reconcilationAfterAgentRestart verifies OVS cleanup and IP release.
+func (data *testData) reconcilationAfterAgentRestart(t *testing.T) error {
+	vlanPod := data.pods[0]
+	iface1 := "eth1"
+
+	logs.Infof("Fetching OVS ports for pod %s before deletion", vlanPod.podName)
+	beforePorts, err := data.getPodPortsAndIPs(vlanPod)
+	if err != nil {
+		return fmt.Errorf("failed to get OVS ports before deletion: %v", err)
+	}
+
+	podIP, _, _ := net.ParseCIDR(beforePorts[iface1][0])
+
+	IPPools, err := data.getIPPoolNames(vlanPod.interfaceNetworks[iface1], "default")
+	if err != nil {
+		return fmt.Errorf("failed to get IPPool: %v", err)
+	}
+
+	if err := data.pauseAndResumeAntreaAgent(); err != nil {
+		return fmt.Errorf("failed to pause Antrea agent: %v", err)
+	}
+
+	logs.Infof("Checking OVS ports after VLAN pod deletion")
+	afterPorts, err := data.getPodPortsAndIPs(vlanPod)
+	if err != nil {
+		logs.Infof("OVS ports for pod %s has been deleted %v", vlanPod.podName, err)
+	}
+
+	if len(afterPorts) != 0 {
+		if afterPorts[iface1][0] == beforePorts[iface1][0] {
+			return fmt.Errorf("Ovs ports still there")
+		}
+	}
+
+	logs.Infof("Checking if VLAN IP is released")
+	if err := data.checkIPReleased(IPPools[0], podIP.String()); err != nil {
+		return fmt.Errorf("VLAN IP release failed: %v", err)
+	}
+
+	return nil
+}
+
+// removePodsAndCheckOVSPorts removes a pod and check if its OVS ports and IPs are released.
+func (data *testData) removePodsAndCheckOVSPorts(t *testing.T) error {
```

Comment:
out of curiosity, I assume that this scenario was already handled correctly prior to this patch?

Reply:
We want to cover the case after Agent restart existing interfaces can be correctly restored and then Pod deletion can delete them and release the IPs.